### PR TITLE
test_store multiworker remove multiprocessing

### PR DESCRIPTION
**Pull request by H-Huang**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#59599 test_store multiworker remove multiprocessing**

This will fix the flakiness for these tests internally when running under TSAN. We don't need multiprocessing since we should restrict the testing to the `wait_for_workers` and `world_size` parameters of the tcp store master store.

Differential Revision: [D28947838](https://our.internmc.facebook.com/intern/diff/D28947838)

### CODE REVIEW

**Comment by H-Huang** [comment]
@H-Huang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28947838).

File: test/distributed/test_store.py [hunk 1]
```diff
@@ -246,51 +241,28 @@ def _test_numkeys_delkeys(self, fs):
...
-        error_message = ""
-        while not messages.empty():
-            error_message += messages.get() + "\n"
-        if any([p.exitcode != 0 for p in processes]):
-            raise RuntimeError(error_message)
-
-    @unittest.skipIf(
-        IS_WINDOWS, "Skip test for windows due to multiprocessing library error when using windows spawn"
-    )
+        world_size = random.randint(5, 10) if world_size else world_size
```
Comment (cbalioglu) [inline test/distributed/test_store.py:256]: Not sure if this ternary operator is correct. Isn't it supposed to be `if world_size == -1`?

**Comment by H-Huang** [comment]
@H-Huang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28947838).

File: test/distributed/test_store.py (hunk 1 above)
Comment (H-Huang) [reply test/distributed/test_store.py]: Good catch, accidentally changed that. Fixed it now, thanks!

**Review by mrshenli — APPROVED** [review]
This looks good to me. Just curious, do we know why creating client in using multiple processes is flaky? I am a little worried, because I assume that's how users will be using it?

**Pull request closed without merging**